core/src/apps/bitcoin/sign_tx/approvers.py

```diff
@@ -154,11 +227,16 @@ def add_change_output(self, txo: TxOutput, script_pubkey: bytes) -> None:
         self.our_weight.add_output(script_pubkey)
         self.group_our_count += 1
 
-    async def add_external_output(self, txo: TxOutput, script_pubkey: bytes) -> None:
-        await super().add_external_output(txo, script_pubkey)
+    async def add_external_output(
+        self,
+        txo: TxOutput,
+        script_pubkey: bytes,
+        orig_txo: Optional[TxOutput] = None,
+    ) -> None:
+        await super().add_external_output(txo, script_pubkey, orig_txo)
```

Comment:
Currently it ignores the information about the original tx. If in the future we choose to support replacement CoinJoin txs, then the host can send the data and older firmware will ignore it while new firmware will use it, so the host doesn't have to figure out whether the firmware supports CoinJoin replacement or not. The reason to support replacement CoinJoin txs would be so that if a CoinJoin tx doesn't go through after we sign, then we can sign another CoinJoin later with the same inputs without depleting the pre-approved fee limit. It's not a strong use-case, so I am not sure that we would implement it, plus the current discussions about the fee policy may make this void.

Reply:
OK, let's keep it as-is.
A point I missed yesterday is that the CoinJoin approver is not affected by the basic approver WRT output confirmation (or indeed confirmation of anything), so ignoring instead of rejecting the replacement info does not affect the results of signing.